hub-agent/templates/deployment-dev-portal.yaml

```diff
@@ -0,0 +1,85 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: hub-agent-dev-portal
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "hub-helm-chart.labels" . | nindent 4 }}
+  {{- with .Values.devPortalDeployment.labels }}
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
+  annotations:
+  {{- with .Values.devPortalDeployment.annotations }}
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  replicas: {{ .Values.devPortalDeployment.replicas }}
+  selector:
+    matchLabels:
+      app: hub-agent
+      component: dev-portal
+  template:
+    metadata:
+      labels:
+        app: hub-agent
+        component: dev-portal
+      {{- with .Values.devPortalDeployment.podLabels }}
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
+      annotations:
+      {{- with .Values.devPortalDeployment.podAnnotations }}
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
+    spec:
+      serviceAccountName: hub-agent-dev-portal
+      automountServiceAccountToken: true
+      {{- with .Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: hub-agent-dev-portal
+          image: {{ template "hub-helm-chart.image-name" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          readinessProbe:
+            httpGet:
+              path: /_ready
+              port: 80
+            initialDelaySeconds: 5
+            periodSeconds: 20
+          livenessProbe:
+            httpGet:
+              path: /_live
+              port: 80
+            initialDelaySeconds: 5
+            periodSeconds: 20
+          resources:
+            {{- with .Values.resources }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+          securityContext:
+            allowPrivilegeEscalation: false
+            readOnlyRootFilesystem: false
```

Comment:
If possible, I would suggest to restrict written data to a specific path.
(And so sets readOnlyRootFilesystem to true with an emptyDir for data)